### PR TITLE
Fix/factories for stETH topup

### PR DESCRIPTION
**Pull request by zuzueeka**
Fix/factories for stETH topup

### CODE REVIEW

File: tests/integration/conftest.py [hunk 1]
```diff
@@ -514,12 +516,18 @@ def _check_top_up_motion_enactment(
             top_up_recipients,
         )
 
-        assert sender_balance == sender_balance_before - spending
+        if top_up_token == lido_contracts.steth:
```
Comment (TheDZhon) [inline tests/integration/conftest.py:519]: can check also `sharesOf` avoiding tolerance issues (in addition, not a replacement)

File: utils/lido.py
Comment (TheDZhon) [inline utils/lido.py:1]: I am not totally convinced about the purpose of these changes (how do we use these roles and accounting oracle further?)

Moreover, the PR title suggests some fixes for the topup factories, while in essence, there is an integration test update, new interfaces, roles, and addresses for the Lido core contracts and

**Review by TheDZhon — APPROVED** [review]
LGTM, though left some additional considerations :pray:

File: tests/integration/conftest.py (hunk 1 above)
Comment (zuzueeka) [reply tests/integration/conftest.py:529]: done

**Pull request merged**